spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/session/SessionStoreType.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2012-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.session;
+
+/**
+ * Supported Spring Session data store types.
+ *
+ * @author Tommy Ludwig
+ * @since 1.4.0
+ */
+public enum SessionStoreType {
+
+	/**
+	 * Redis backed sessions.
+	 */
+	REDIS,
+
+	/**
+	 * Hazelcast backed sessions.
+	 */
+	HAZELCAST,
+
+	/**
+	 * Simple in-memory map of sessions.
+	 */
+	SIMPLE,
```

Comment:
@rwinch I am not a native speaker: you lost me with the `SLIGHTLY_TRICKY` part to be honest. Is that ironic? 

`SIMPLE` means exactly what that store type is IMO. I favour consistency above all and `SIMPLE` is [exactly the name we chosen for the "hash map" cache manager option](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/CacheType.java#L77).